src/main.cpp

```diff
@@ -102,79 +178,117 @@ void wrongDataState(){
  * Average methods
  **/
 
+#ifndef SENSIRION
 void saveDataForAverage(unsigned int pm25, unsigned int pm10){
   v25.push_back(pm25);
   v10.push_back(pm10);
 }
+#else
+void saveDataForAverage(unsigned int pm25, unsigned int pm10, float pm25f){
+  v25.push_back(pm25);
+  v10.push_back(pm10);
+  v25f.push_back(pm25f);
+}
+#endif
 
 unsigned int getPM25Average(){
-  unsigned int pm25_average = accumulate( v25.begin(), v25.end(), 0.0)/v25.size();
+  unsigned int pm25_average = round(accumulate(v25.begin(), v25.end(), 0.0) / v25.size());
   v25.clear();
   return pm25_average;
 }
 
 unsigned int getPM10Average(){
-  unsigned int pm10_average = accumulate( v10.begin(), v10.end(), 0.0)/v10.size();
+  unsigned int pm10_average = round(accumulate(v10.begin(), v10.end(), 0.0) / v10.size());
   v10.clear();
   return pm10_average;
 }
 
+#ifdef SENSIRION
+float getPM25fAverage(){
+  float pm25f_average = accumulate(v25f.begin(), v25f.end(), 0.0) / v25f.size();
+  v25f.clear();
+  cfg.spd = pm25f_average;
+  return pm25f_average;
+}
+#endif
+
 void averageLoop(){
   if (v25.size() >= cfg.stime){
     apm25 = getPM25Average();  // global var for display
     apm10 = getPM10Average();
+  #ifdef SENSIRION
+    apm25f = getPM25fAverage();
+  #endif
   }
 }
 
-char getLoaderChar(){ 
+char getLoaderChar(){
   char loader[] = {'/','|','\\','-'};
   return loader[random(0,4)];
 }
 
-void showValues(int pm25, int pm10)
-{
+#ifndef SENSIRION
+void showValues(int pm25, int pm10){
   gui.displaySensorAverage(apm25); // it was calculated on bleLoop()
   gui.displaySensorData(pm25, pm10, chargeLevel, humi, temp, rssi);
   gui.displayLiveIcon();
   saveDataForAverage(pm25, pm10);
+  WrongSerialData = false;
```

Comment:
Why is the reason for have a global variable here?

Reply:
deleted pm25f
WrongSerialData is a flag for correct a bug with the TTGO_TQ that is rest continuously after each power on or reset